src/Semantic/Resolution.hs

```diff
@@ -51,7 +50,6 @@ nodeJSResolutionMap rootDir prop excludeDirs = do
 
     lookupProp :: FilePath -> Text -> Object -> Maybe (Map FilePath FilePath)
     lookupProp path k res = flip parseMaybe res $ \obj -> Map.singleton relPkgDotJSONPath . relEntryPath <$> obj .: k
-      -- JZ:TODO: shall be refactored
       where relPkgDotJSONPath = makeRelative (Path.toString rootDir) path
```

Comment:
I think we can refactor relPkgDotJSONPath and some following variables with pathtype.  But this PR is big enough.  I would like to refactor it in some other small PRs.